tests/lit-tests/loop_nounroll.ispc

```diff
@@ -0,0 +1,99 @@
+// Test to check '#pragma nounroll' functionality for different loop statemnts.
+
+// RUN: %{ispc} %s --target=avx2-i32x8 --nowrap -O0 --emit-llvm-text -o - | FileCheck %s -check-prefix=CHECKO0
+// RUN: %{ispc} %s --target=avx2-i32x8 --nowrap -O2 --emit-llvm-text -o - | FileCheck %s -check-prefix=CHECKO2
+// REQUIRES: X86_ENABLED
+
+// CHECKO0: br label %for_test, !llvm.loop
+
+// CHECKO2: call void @goo_for___uni
+// CHECKO2-NOT: call void @goo_for___uni
+
+extern void goo_for(uniform int);
```

Comment:
When there are several functions in the same test, it's good to verify that the check matches the text of specific function. I.e. you need to match beginning of the function first: ``define void @goo_for___``

Reply:
The names are different for all function. They all start with goo..but it's goo_for vs goo_foreach_unique etc. But yeah..can match the define too

Reply:
Done

Reply:
You need a check for function start for O0 as well.

Reply:
Added